### PR TITLE
feature/CB2-2879: Authentication

### PR DESCRIPTION
**Pull request by NiallBunting**
## feature/CB2-2879
_One line description_
[link to ticket number]()

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [x] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by Ian-Hodges — APPROVED** [review]

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,22 +1,96 @@
...
+    cache: {
+      cacheLocation: BrowserCacheLocation.LocalStorage,
+      storeAuthStateInCookie: true,
+    },
+  });
+}
+
+export function MSALInterceptorConfigFactory(): MsalInterceptorConfiguration {
+  const protectedResourceMap = new Map<string, Array<string>>();
+  protectedResourceMap.set("https://graph.microsoft.com/v1.0/me", ["user.read"]);
```
Comment (premghinde) [inline src/app/app.module.ts:42]: Is eslint not pickup on the double quotes?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,22 +1,96 @@
...
 import { FooterComponent } from './layout/footer/footer.component';
+import { environment } from '../environments/environment';
+import { UserService } from './user-service';
+
+export function MSALInstanceFactory(): IPublicClientApplication {
+  return new PublicClientApplication({
+    auth: {
+        clientId: environment.VTM_CLIENT_ID,
+        authority: environment.VTM_AUTHORITY_ID,
+        redirectUri: '/',
```
Comment (premghinde) [inline src/app/app.module.ts:30]: add the uri to configuration please

File: src/app/user-service.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { MsalService, MsalBroadcastService, } from "@azure/msal-angular";
```
Comment (premghinde) [inline src/app/user-service.ts:1]: inconsistent indentations in this file should be flagged by the linter

File: src/app/app.module.ts (hunk 1 above)
Comment (aaron-downing) [reply src/app/app.module.ts:42]: I'm currently working on getting eslint working properly :)

Comment (NiallBunting) [reply src/app/app.module.ts:42]: I'm just going to leave these linter comments, as waiting on the linter to work.

**Pull request merged**